hr_timesheet_holiday/models/analytic_line.py

```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Sunflower IT <http://sunflowerweb.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning as UserError
+
+class AnalyticLine(models.Model):
+    """Restrict edit and delete on analytic lines generated by holidays"""
+    _inherit = 'account.analytic.line'
+
+    _timesheet_ids = fields.One2many('hr.analytic.timesheet', 'line_id')
+
+    @api.multi
+    def write(self, vals):
+        if not self.env.context.get('force_write', False):
+            for line in self:
+                if line.account_id.is_leave_account \
+                        and line._timesheet_ids.leave_id:
+                    raise UserError(_("This line is protected against editing"
+                        " because it was created automatically by a leave request."
+                        " Please edit the leave request instead."))
+        return super(AnalyticLine, self).write(vals)
+
+    @api.multi
+    def unlink(self):
+        if not self.env.context.get('force_unlink', False):
+            for line in self:
+                if line.account_id.is_leave_account \
+                        and line._timesheet_ids.leave_id:
+                    raise UserError(_("This line is protected against removal"
+                        " because it was created automatically by a leave request."
+                        " Please edit the leave request instead."))
+        return super(AnalyticLine, self).unlink()
```

Comment:
better implement both of them as one `@api.constrains` function on `account_id`, `unit_amount`, `amount`